homeassistant/components/history/__init__.py

```diff
@@ -127,9 +132,11 @@ def state_changes_during_period(hass, start_time, end_time=None, entity_id=None)
 
         entity_ids = [entity_id] if entity_id is not None else None
 
-        states = execute(query.order_by(States.last_updated))
+        states = execute(
+            query.order_by(States.entity_id, States.last_updated), to_native=False
```

Comment:
Since `to_native` is such a performance killer, we should defeault it to `False` in a future PR.

Reply:
Queued in my todo list.

Reply:
done here https://github.com/home-assistant/core/pull/36938